### DISCUSSION TITLE
Has the CommunityToolkit team considered releasing the common analyzer code as a package?

### DISCUSSION THREAD

**Discussion by kmgallahan** [CommunityToolkit/dotnet#686] [Q&A] [answered]
As per the title, the code I'm referring to would be that which is reusable/common and could be applied to any analyzer project. 

That would include such things as the contents of `CommunityToolkit.Mvvm.SourceGenerators`'s Helpers, Extensions, and Models namespaces.

I ask because I just created an analyzer (source generator with diagnostic) and found the code in those namespaces quite useful, yet internal.

**Comment by Sergio0694** [comment, marked as answer]
I would've loved to do this, but unfortunately you cannot reference NuGet packages from source generators and analyzers at this time. I mean you can (kinda) but it's very clunky, can cause problems, and there's no official tooling support for it. When/if that was available, it's definitely something we'd consider 🙂

**Reply by kmgallahan** [reply]
I see, perhaps in the future then 👍
